Final Project

ameerhamzakhalid opened the pull request:
### **In this capstone project,**

I have built a web app for getting Air pollution data and weather data for different cities
here is the [LINK](https://drive.google.com/file/d/1fKQU08Cj6YhonRVwMOJogjP6-aAaPlDw/view?usp=share_link) to my presentation.
I achieved the following:

-     To get the air pollution data user can navigate to the details page.
-     Created Details page, and retrieve data from open weather and Air pollution.
-     When a user clicks the back button he is taken back to the home page.
-     Created Unit tests for reducers.
-     created test cases for react components.


I have followed the below requirements:

-     The project is a single-page application (SPA) built with React and Redux.
-     The data retrieved from the API should be stored in the Redux store.
-     Should filter the data that you retrieve from the API using a Filter stateless component.
-     Every page (the main page and the pages for each item) should have a unique route within the SPA.
-     The project should be deployed and accessible online (add a link to the online version of your app to your README file).

ameerhamzakhalid committed:
finalized my capstone project

I have built a web app for getting Air pollution data and weather data for different cities

    The project is a single-page application (SPA) built with React and Redux.
    The data retrieved from the API should be stored in the Redux store.
    Should filter the data that you retrieve from the API using a Filter stateless component.
    Every page (the main page and the pages for each item) should have a unique route within the SPA.
    The project should be deployed and accessible online (add a link to the online version of your app to your README file).

ameerhamzakhalid committed:
updatd token

ameerhamzakhalid committed:
updated api deployment

ameerhamzakhalid committed:
updated readme

added live demo link

omarsalem7 commented on README.md:
```diff
@@ -1 +1,166 @@
+
+I would like to thank microverse for the great resources shared.
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
```
- [x] Kindly give appropriate credit to [Nelson Sakwa](https://www.behance.net/sakwadesignstudio) the [author of the original design](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), as required by the [Creative Commons License](https://creativecommons.org/licenses/).

ameerhamzakhalid replied:
done

omarsalem7 commented on README.md:
```diff
@@ -1 +1,166 @@
+
+## 📝 License <a name="license"></a>
+
+This project is [MIT](./LICENCE) licensed.
```
- [ ] Kindly fix the broken MIT license link in the readme file and add your own license.

ameerhamzakhalid replied:
@omarsalem7 done sir!

omarsalem7 commented on src/components/Cities.js:
```diff
@@ -0,0 +1,98 @@
+          ;
+        </select>
+      </div>
+      <div className="cities">
+        {cities.length > 0 ? cities.map((c) => (c === 'init' ? <p Key="m" className="message">Select Your State</p> : (
+          <Link key={c.id} className="city-card" to={`/Details/${c.coord.lat}/${c.coord.lon}`} state={c}>
+            <p className="city-name">{c.name}</p>
+            <div className="city-info">
+              <p>
+                weather :
+                {` ${c.weather}`}
+              </p>
+              <p>
+                temperature :
+                {` ${c.temp} C°`}
+              </p>
+              <p>
+                feels like :
+                {` ${c.feel} C°`}
+              </p>
+            </div>
+          </Link>
+        ))) : (
+          <p className="message">Sorry! no cities found</p>
+        )}
+      </div>
```
- [x] [OPTIONAL] It will be better if you add a loading indicator when loading cities to indicate to users that that app is loading cities.

ameerhamzakhalid replied:
Hi @omarsalem7 thank you so much for your kind optional suggestion will consider it in future because of running out of time 😊

omarsalem7 reviewed (changes requested):
Hi @ameerhamzakhalid 😊, 

Good job so far! 👏
You have done well reaching this point of the milestone, but there are some issues that you still need to work on to go to the next project but you are almost there! 

![you-are-near](https://media.giphy.com/media/j7vckgz15cAYmxatzB/giphy.gif)

## Highlights
-  The application deployed using Netlify properly ✔️
- The project is a SPA using `react-router-dom` to serve each page without reloading.✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
- Check the comment under the review

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

ameerhamzakhalid committed:
update readme

I added acknowledgments to Nelson Sakwa and fixed MIT license link as suggested by the previous reviewer

Davidosky007 commented on src/__TESTS__/reducers.test.js:
```diff
@@ -0,0 +1,60 @@
+      loading: true,
+      error: null,
+    };
+    const result = citiesReducer(undefined, { type: '' });
```
## OPTIONAL 
* I strongly suggest you fix this error i get when i run your test, i mark this as optional because the previous CR missed this, But you can fix it in your free time to make your project awesome 💯  
![image](https://user-images.githubusercontent.com/52497246/218010941-46a1c991-d59e-495a-a13c-7addc35ee4a1.png)

Davidosky007 reviewed (approved):
Hi @ameerhamzakhalid 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* Live link works well 👍 
* Good looking readme file 👍 
* Professional video presentation 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
